src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReportService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String CATEGORIES = "fruit,quantity";
+
+    @Override
+    public List<String> generateReport(Map<String, Integer> fruitCounts) {
+        if (fruitCounts == null) {
+            throw new NullPointerException("Can't generate report for null data");
+        } else {
```

Comment:
```suggestion
    if (fruitCounts == null) {
        throw new IllegalArgumentException("Can't generate report for null data");
    }
```
We don't deal with NPE explicitly - it happens due to programmer's mistakes, and should be avoided. + We don't need else after throwing/returning